lib/miq_automation_engine/engine/miq_ae_event.rb

```diff
@@ -138,36 +138,44 @@ def self.call_automate(obj, attrs, instance_name, options = {})
     end
   end
 
-  def self.provider_event_target(event)
-    event.vm_or_template || event.ext_management_system
-  end
-
-  def self.miq_event_target(target)
-    case target
-    when VmOrTemplate, MiqServer
-      target
+  def self.stream_target(object)
+    case object
+    when EmsEvent
+      object.vm_or_template || object.ext_management_system
+    when VmOrTemplate, MiqServer, Service, ExtManagementSystem, MiqRequest
+      object
     else
-      target.ext_management_system
+      object.ext_management_system
     end
   end
 
   def self.automate_user_ids(object)
-    target = case object
-             when EmsEvent
-               provider_event_target(object)
-             else
-               miq_event_target(object)
-             end
-
-    user  = User.super_admin
-    group = target.kind_of?(MiqServer) ? user.current_group : target.miq_group
-    raise "A group is needed to raise events" unless group
-
-    tenant = group.current_tenant
-    user = target.evm_owner if target.kind_of?(VmOrTemplate) && target.evm_owner && target.evm_owner.miq_groups.include?(group)
+    target = stream_target(object)
+    case target
+    when VmOrTemplate, Service
+      group = target.miq_group
+      user = target.evm_owner
+      user = User.super_admin if user.nil? || !user.miq_groups.include?(group)
+      tenant = target.tenant
+    when ExtManagementSystem
+      user = User.super_admin
+      # TODO: uncomment when default group for tenant is merged
+      # group = target.tenant.default_miq_group
+      group = user.current_group # for now
+      tenant = target.tenant
+    when MiqServer
+      user = User.super_admin
+      group = user.current_group
+      tenant = user.current_tenant
+    when MiqRequest
+      user = target.get_user
+      group = user.current_group
+      tenant = user.current_tenant
+    end
 
+    raise "A group is needed to raise events" unless group
```

Comment:
@lfu Can we add some detail about the event to this message.  Would be nice to have the event class/ID and event_type listed.

Something like:
`raise "A group is needed to raise an event.  #{self.class.name} id:#{id} event_type: #{event_type}" unless group`